### PR TITLE
Metric: bus factor

### PR DESCRIPTION
**Pull request by vdudhaiy**
Wrote the bus factor function along with sample output (freeCodeCamp repo)

### CODE REVIEW

File: apis/git_api.py [hunk 1]
```diff
@@ -73,22 +76,6 @@ def get_contributors(owner: str, repo: str) -> List[Dict[str, Any]]:
     response = make_request(url, headers)
     return response.json()
 
-def get_commit_history(owner: str, repo: str) -> List[Dict[str, Any]]:
```
Comment (golovenchits) [inline apis/git_api.py:76]: Why delete this function?

**Comment by vdudhaiy** [comment]
> When I run the metric on one of my repos, which only I have contributed to, I get a 1. Wouldn't the bus factor have to be 0 in that case, since only 1 person ever contributed?

Good catch, yeah, the bus factor is a little more messed up when dealing with repos that have fewer contributors. I think the 1 contributor case might have to be like an exception or a special case scenario. 

There are some other pitfalls that I have also encountered so I'm gonna spend some time trying to resolve those too. Apologies for the premature pull request.

**Comment by vdudhaiy** [comment]
Made the requested changes:
- Restored get_commit_history()
- Added edge case (1 contributor only)
- Added test cases for each case (large open source repo, small open source repo, repo with 2 contributors, repo with 1 contributor"

**Comment by golovenchits** [comment]
> **One concern is as follows:** Currently, we're ranking the contributors based on number of commits but we're not taking into account how long ago those commits were made. For instance, if a contributor made a large number of contributions and they are ranked among the contributors that amount to 50% of contributions authored, but they haven't made any contributions in a while, are they still to be considered a critical contributor? Let me know your thoughts.

I think we don't need to worry about that, we should just keep it simple IMO

**Review by golovenchits — APPROVED** [review]

**Review by Murad777-phy — APPROVED** [review]
